infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/SessionManager.java

```diff
@@ -443,7 +446,15 @@ private void removeStaleSessionOnLeadershipAcquire() {
      * @return set of sessions for which this cluster is the sink
      */
     public Set<LogReplicationSession> getIncomingSessions() {
-        return incomingSessions;
+        if (replicationContext.getIsLeader().get()) {
+            return incomingSessions;
+        }
+        Set<LogReplicationSession> sessionsToReturn;
+        try (TxnContext txn = metadataManager.getTxnContext()) {
+            sessionsToReturn = txn.keySet(metadataManager.getMetadataTable());
```

Comment:
sorry that this did not occur to me earlier, but should we also filter the sessions where the model == FULL_TABLE?  In a real topology, since we do not support other models yet, returning all incoming and outgoing sessions can lead to unexpected behavior

Reply:
I am thinking of removing the support for other models all together...
Also, I think the caller of this function should filter out the FULL_TABLE...we should return all the sessions created.